### PR TITLE
.NET SDK Examples

### PR DESCRIPTION
**Pull request by sohail-aspose**
I have written SDK Examples, and they are in the "Examples" folder. Please merge my branch into the master branch.

### CODE REVIEW

File: Examples/AsposeImagingCloudSDKExamples/CropImage.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿using Aspose.Imaging.Cloud.Sdk.Api;
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/CropImage.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/CropImage.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AsposeImagingCloudSDKExamples
+{
+    class CropImage : ImagingBase
+    {
+        // Crop an existing image
+        public void cropImageInCloud()
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/CropImage.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/CropImage.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            string format = "jpg"; // Resulting image format. Currently, BMP, PSD, JPG, TIFF, GIF, PNG, J2K and WEBP are supported.
+            int? x = 10;
+            int? y = 10;
+            int? width = 100;
+            int? height = 150;
+            string storage = null; // We are using default Cloud Storage
+            string outPath = null; // Path to updated file (if this is empty, response contains streamed image)
+
+            var request = new CreateCroppedImageRequest(inputImageStream, format, x, y, width, height, outPath, storage);
+            Stream updatedImage = this.ImagingApi.CreateCroppedImage(request);
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/CropImage.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/CropImage.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            int? width = 100;
+            int? height = 150;
+            string folder = null; // Input file is saved at the root of the storage
+            string storage = null; // We are using default Cloud Storage
+
+            var request = new CropImageRequest(fileName, format, x, y, width, height, folder, storage);
+            Stream updatedImage = this.ImagingApi.CropImage(request);
+
+            // Save updated image to local storage
+            using (var fileStream = File.Create(ImagingBase.PathToDataFiles + "Watermark_out." + format))
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/CropImage.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/ExportImage.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+            // Upload local image to Cloud Storage
+            using (FileStream localInputImage = File.OpenRead(ImagingBase.PathToDataFiles + fileName))
+            {
+                var uploadFileRequest = new UploadFileRequest(fileName, localInputImage);
+                FilesUploadResult result = this.ImagingApi.UploadFile(uploadFileRequest);
+            }
+
+            // Output formats could be one of the following:
+            // BMP,	GIF, DJVU, WMF, EMF, JPEG, JPEG2000, PSD, TIFF, WEBP, PNG, DICOM, CDR, CMX, ODG, DNG, SVG and PDF
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/ExportImage.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/ai/ImagingAIBase.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AsposeImagingCloudSDKExamples
+{
+    class ImagingAIBase : ImagingBase
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/ai/ImagingAIBase.cs]: Completed. Should I incorporate a call to create "ImagingAPI" instance as well in each Example?
public ImagingBase()
{
    this.ImagingApi = new ImagingApi(MyAppKey, MyAppSid);
}

File: Examples/AsposeImagingCloudSDKExamples/ai/CompareImages.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace AsposeImagingCloudSDKExamples.ai
+{
+    class CompareImages : ImagingAIBase
+    {
+        private const string ComparableImage = "ComparableImage.jpg";
+        private const string ComparingImageSimilarLess15 = "ComparingImageSimilar15.jpg";
+        private const string ComparingImageSimilarMore75 = "ComparingImageSimilar75.jpg";
+
+        // Compare two images. Image data may be passed as zero-indexed multipart/form-data content or as raw body stream.
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/ai/CompareImages.cs]: Completed.

File: Examples/AsposeImagingCloudSDKExamples/ExportImage.cs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AsposeImagingCloudSDKExamples
+{
+    class ExportImage : ImagingBase
+    {
+        public void exportImageToAnotherFormat()
```
[the remark this answers is not in the document]
Comment (sohail-aspose) [reply Examples/AsposeImagingCloudSDKExamples/ExportImage.cs]: Completed.

**Comment by sohail-aspose** [comment]
@MaxCrank I have made the suggested changes. Please review them, and let me know if I have to make any other change.

File: Examples/AsposeImagingCloudSDKExamples/ai/ImagingAIBase.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (aifeigin) [reply Examples/AsposeImagingCloudSDKExamples/ai/ImagingAIBase.cs]: _Should I incorporate a call to create "ImagingAPI" instance as well in each Example?_
It's not necessary, as it's clear from implementation.

But probably you could add parameter title to ImagingBase constructor and output it to console, making output more user friendly.

**Comment by MaxCrank** [comment]
@sohail-aspose I'll finish the work on this pull request tomorrow, finally having some free time. Thank you for your work and sorry for responding so late - I had to go on vacation due to personal reasons.

@aifeigin FYI

**Pull request merged**